lib/IRGen/GenDecl.cpp

```diff
@@ -2959,7 +2959,7 @@ IRGenModule::getAddrOfMetaclassObject(ClassDecl *decl,
 llvm::Function *
 IRGenModule::getAddrOfObjCMetadataUpdateFunction(ClassDecl *classDecl,
                                                  ForDefinition_t forDefinition) {
-  IRGen.noteUseOfTypeMetadata(classDecl);
```

Comment:
Is this not necessary?

Reply:
I don't think so, no. While it doesn't hurt, it's confusing to see it there when the only time we reference the update callback is while we're _already_ emitting the metadata.